concrete-core/src/specification/engines/lwe_ciphertext_inplace_extraction.rs

```diff
@@ -0,0 +1,51 @@
+use super::engine_error;
+use crate::specification::engines::AbstractEngine;
+use crate::specification::entities::{GlweCiphertextEntity, LweCiphertextEntity};
+use concrete_commons::parameters::MonomialDegree;
+
+engine_error! {
+    LweCiphertextInplaceExtractionError for LweCiphertextInplaceExtractionEngine@
+    SizeMismatch => "The sizes of the output lwe and the input glwe must be compatible.",
+    MonomialDegreeTooLarge => "The monomial degree must be lower than the glwe polynomial size."
+}
+
+/// A trait for engines extracting (inplace) lwe ciphertext from glwe ciphertexts.
+///
+/// # Semantics
+///
+/// This [inplace](super#operation-semantics) operation fills the `output` lwe ciphertext with the
+/// sample extraction of the `nth` coefficients of the `input` glwe ciphertext.
+///
+/// # Formal definition
+///
+/// This operation is usually referred to as a _sample extract_ in the literature.
```

Comment:
It is also called sample extraction literally 4 lines above. Which means that someone who does not come from the literature is not closer to understand what this operation does. 
That being said, I find the definition clear enough, but then again I know what a sample extract is. In doubt, I would get rid of the word "sample" in the `Semantics` part and add something like "This means that `output` will become an encryption of the `nth` coefficient of the message encrypted by `input`" or something more intelligible.

Reply:
Good idea. I guess it is better to leave all mentions to _sample extraction_ under the __Formal Definition__ section.

Reply:
Actually I also expect the __Formal Definition__ section to be way more informative on what this actually is in the future :)